cmd/api/src/api/v2/agi.go

```diff
@@ -258,8 +258,17 @@ func (s Resources) UpdateAssetGroupSelectors(response http.ResponseWriter, reque
 		if result, err := s.DB.UpdateAssetGroupSelectors(*ctx.FromRequest(request), assetGroup, selectorSpecs, false); err != nil {
 			api.HandleDatabaseError(request, response, err)
 		} else {
-			// When T0 asset group selectors are modified we must trigger analysis
+			// update the asset group tags immediately
+			if err = s.GraphQuery.ClearSystemTags(request.Context()); err != nil {
+				log.Warnf("failed clearing asset group tags; will be retried upon next analysis run: %v", err)
+			}
+
+			if err = s.GraphQuery.UpdateAssetGroupIsolationTags(request.Context(), s.DB); err != nil {
+				log.Warnf("failed updating asset group tags; will be retried upon next analysis run: %v", err)
+			}
```

Comment:
I notice that any modification to an asset group causes all members of other asset groups to disappear. I think only the nodes associated with a selector need to be modified/tagged.